lib/ansible/modules/network/iosxr/iosxr_config.py

```diff
@@ -251,7 +251,7 @@ def copy_file_to_node(module):
     """
     src = '/tmp/ansible_config.txt'
     file = open(src, 'wb')
-    file.write(module.params['src'])
+    file.write(to_bytes(module.params['src']))
```

Comment:
By default `to_bytes` uses `errors='surrogate_then_replace'` which can cause the resultant contents to be incorrect.  You probably want an explicit `errors='surrogate_or_strict'`.

Just as a note, in case you want to take this on, it looks like this whole function could use a rewrite.  But I'm in no way suggesting you should.

1. Should use a context manager
1. Should not use a hardcoded `/tmp/ansible_config.txt` as the source, should use a tmpdir/tmpfile instead
1. does `copy_file` clean up the `src`?

Reply:
Great feedback, will update to use errors='surrogate_or_strict' as you suggest. I also agree, this function could use a clean up, I'll raise it with the team and add suggest we add it to backlog.

Reply:
Great feedback, will update to use errors='surrogate_or_strict' as you suggest. I also agree, this function could use a clean up, I'll raise it with the team and add suggest we add it to backlog.